client/app/bundles/course/assessment/submission/reducers/answerFlags/index.ts

```diff
@@ -58,11 +58,8 @@ export const answerFlagsSlice = createSlice({
       action: PayloadAction<{
         answer: { id: number | string };
         savingStatus: AnswerFlagData['savingStatus'];
-        isStaleAnswer?: boolean;
       }>,
     ) => {
-      if (action.payload.isStaleAnswer) return;
```

Comment:
Why is this removed?

Reply:
the previous notion of this stale answer is only when there's a race condition within the same action (for example, while you're typing and then the API for saving is sent, and then when it goes back the redux store for clientVersion is updated -> another API call will be received and this answerFlag would be updated eventually).

However, with now the additional action that might trigger API call (clicking other bubble), having this line will possibly make the state to be consistent in "Saving" mode, even though the answer has been saved properly.

So my approach here is basically if the staleAnswer is met, then we revert the savingStatus to what it was before the saving takes place.. it might still give the savingStatus the persistent "Saving" state, but the occurences will be much less likely

Reply:
> it might still give the savingStatus the persistent "Saving" state, but the occurences will be much less likely
1. I managed to reproduce a locked "Saving" state when switching between bubbles. How can we prevent this?
2. When switching bubbles, we should immediately flush the outstanding by dispatching a save (done), as well as cancel the debounced saved (not yet done).